packages/create-react-app-demo/src/recipe/page.tsx

```diff
@@ -78,33 +79,31 @@ export default function RecipeWrapper() {
       <InputPrompt label="What would you like a recipe for?" value={query} setValue={setQuery} />
 
       <ResultContainer title={`AI comes up with a recipe for ${query}`}>
-        <AI renderPassedReactComponents>
-          <ChatCompletion temperature={1}>
-            <SystemMessage>
-              You are an AI who is an expert chef and also an expert UI designer. The user will ask you for a recipe.
-              Your response must be structured using a set of React components. Here are the React components, and an
-              example of how they should be used:
-              <Recipe>the entire recipe contents</Recipe>
-              <RecipeTitle>the title of your recipe</RecipeTitle>
-              <RecipeInstructionList>
-                <RecipeInstructionListItem>the first instruction</RecipeInstructionListItem>
-              </RecipeInstructionList>
-              <RecipeIngredientList>
-                <RecipeIngredientListItem>the first ingredient</RecipeIngredientListItem>
-              </RecipeIngredientList>
-              Every child of a RecipeInstructionList should be a RecipeInstructionListItem. Every child of a
-              RecipeIngredientList should be a RecipeIngredientListItem. Respond with a JSON object that encodes your
-              UI. The JSON object should match this TypeScript interface: interface Element {'{'}
-              name: string; children: (string | Element)[]
-              {'}'}
-              For example:
-              {'{'}
-              "name": "Recipe", "children": [{'{'}"name": "RecipeTitle", "children": ["My Recipe"]{'}'}
-              "my description" ]{'}'}. Respond with only the JSON. Do not include with an explanatory suffix or prefix.
-            </SystemMessage>
-            <UserMessage>Give me a recipe for {query}. Respond with only the JSON.</UserMessage>
-          </ChatCompletion>
-        </AI>
+        <AI.jsx>
+          <UICompletion
+            example={
```

Comment:
Splitting out the example as a separate prop, rather than the user mixing it in with everything else, is an interesting approach. I like it.